packages/contracts/contracts/Router.sol

```diff
@@ -0,0 +1,220 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.4;
+
+import "./interfaces/ITransactionManager.sol";
+import "./lib/LibAsset.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+
+contract Router is Ownable {
+  address public immutable routerFactory;
+
+  ITransactionManager public transactionManager;
+
+  uint256 private chainId;
+
+  address public recipient;
+
+  address public routerSigner;
+
+  struct SignedPrepareData {
+    ITransactionManager.PrepareArgs args;
+    address routerRelayerFeeAsset;
+    uint256 routerRelayerFee;
+    uint256 chainId; // For domain separation
```

Comment:
if you want to include proper domain separation, you also need to include the address of this contract. same to all the other signed type structs here

Reply:
address on all the chains are going to be the same, so it won't help in domain separation.